confgenerator/confmerger.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package confgenerator provides functions to generate subagents configuration from unified agent.
+package confgenerator
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+var (
+	builtInConf = map[string]string{
```

Comment:
Consider using https://golang.org/pkg/embed/ for this.

Reply:
If the Linux and Windows default configs are 90% similar, is it worth tracking the differences rather than maintaining two completely separate strings (that could fall out of sync)?

Reply:
Yup, this is something we can definitely improve.

Reply:
Added a TODO. I'll work on it while this PR is being reviewed. But it's not a GA blocker, so if the PR is ready to ship before I get there, I will send a followup PR.

Reply:
Done